### PR TITLE
Fix missing logger event and add ma_wait_time for Iterator

### PR DESCRIPTION
**Pull request by mensfeld**
Adds missing instrumentation event hook for LoggerListener and adds ability to customize max_wait_time on the iterator.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
 # Karafka framework changelog
 
 ## 2.1.7 (Unreleased)
+- [Improvement] Add `max_wait_time` option to the Iterator.
 - [Fix] Fix a case where Iterator with per partition offsets with negative lookups would go below the number of available messages.
 - [Fix] Remove unused constant from Admin module.
+- [Fix] Add missing `connection.client.rebalance_callback.error` `LoggerListener` instrumentation hook.
```
Comment (nijikon) [inline CHANGELOG.md:7]: Add missing `connection.client.rebalance_callback.error` to `LoggerListener` instrumentation hook.

File: lib/karafka/instrumentation/logger_listener.rb [hunk 2]
```diff
@@ -277,6 +277,9 @@ def on_error_occurred(event)
         when 'connection.client.poll.error'
           error "Data polling error occurred: #{error}"
           error details
+        when 'connection.client.rebalance_callback.error'
+          error "Rebalance callack error occurred: #{error}"
```
Comment (nijikon) [inline lib/karafka/instrumentation/logger_listener.rb:281]: Rebalance callback error occurred

**Comment by mensfeld** [comment]
@nijikon fixed thanks!

**Review by nijikon — APPROVED** [review]
LGTM

**Pull request merged**